subprojects/gradle/test-project/src/main/kotlin/com/avito/test/gradle/TestProjectGenerator.kt

```diff
@@ -157,35 +165,39 @@ buildCache {
 
 private fun generateIncludes(modules: List<Module>, prefix: String): String =
     modules.joinToString(separator = "\n") {
-        "include('$prefix:${it.name}')" + "\n" + generateIncludes(it.modules, "$prefix:${it.name}")
+        "include(\"$prefix:${it.name}\")" + "\n" + generateIncludes(it.modules, "$prefix:${it.name}")
     }
 
-private fun repositories(): String = if (artifactoryUrl == null) {
+private fun repositories(useKts: Boolean): String = if (artifactoryUrl == null) {
     """
-    repositories {
-        mavenCentral()
-        gradlePluginPortal()
-        google()
-    }
-    """.trimIndent()
+    |repositories {
+    |    mavenCentral()
+    |    gradlePluginPortal()
+    |    google()
+    |}
+    """.trimMargin()
 } else {
     """
-    repositories {
-        maven {
-            name = "Proxy for https://repo1.maven.org/maven2"
-            setUrl("$artifactoryUrl/mavenCentral")
-            allowInsecureProtocol = true
-        }
-        maven {
-            name = "Proxy for https://dl.google.com/dl/android/maven2/"
-            setUrl("$artifactoryUrl/google-android")
-            allowInsecureProtocol = true
-        }
-        maven {
-            name = "Proxy for https://plugins.gradle.org/m2/"
-            setUrl("$artifactoryUrl/gradle-plugins")
-            allowInsecureProtocol = true
-        }
+    |repositories {
+    |    ${artifactoryProxyMavenRepo("Proxy for https://repo1.maven.org/maven2", "mavenCentral", useKts)}
+    |    ${artifactoryProxyMavenRepo("Proxy for https://dl.google.com/dl/android/maven2/", "google-android", useKts)}
+    |    ${artifactoryProxyMavenRepo("Proxy for https://plugins.gradle.org/m2/", "gradle-plugins", useKts)}
+    |}
+    """.trimMargin()
+}
+
+private fun artifactoryProxyMavenRepo(name: String, repo: String, useKts: Boolean): String {
```

Comment:
`useKts` is available at the class level. We don't need to pass it by arguments.